internal/services/appservice/function_app_flex_consumption_resource.go

```diff
@@ -300,6 +368,231 @@ func (r FunctionAppFlexConsumptionResource) Arguments() map[string]*pluginsdk.Sc
 
 		"tags": commonschema.Tags(),
 	}
+	if !features.FivePointOh() {
+		schema["storage_account_name"] = &pluginsdk.Schema{
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			Computed:     true,
+			ValidateFunc: storageValidate.StorageAccountName,
+			Description:  "The backend storage account name which will be used by this Function App.",
+		}
+
+		schema["storage_account_access_key"] = &pluginsdk.Schema{
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			Computed:     true,
+			Sensitive:    true,
+			ValidateFunc: validation.NoZeroValues,
+			ConflictsWith: []string{
+				"storage_account_uses_managed_identity",
+				"storage_account_key_vault_secret_id",
+			},
+			RequiredWith: []string{
+				"storage_account_name",
+			},
+			Description: "The access key which will be used to access the storage account for the Function App.",
+		}
+
+		schema["storage_account_uses_managed_identity"] = &pluginsdk.Schema{
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+			Default:  false,
+			ConflictsWith: []string{
+				"storage_account_access_key",
+				"storage_account_key_vault_secret_id",
+			},
+			RequiredWith: []string{
+				"storage_account_name",
+			},
+			Description: "Should the Function App use its Managed Identity to access storage?",
+		}
+
+		schema["storage_account_key_vault_secret_id"] = &pluginsdk.Schema{
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			Computed:     true,
+			ValidateFunc: kvValidate.NestedItemIdWithOptionalVersion,
+			ConflictsWith: []string{
+				"storage_account_name",
+			},
+			Description: "The Key Vault Secret ID, including version, that contains the Connection String to connect to the storage account for this Function App.",
+		}
+
+		schema["storage_container_type"] = &pluginsdk.Schema{
```

Comment:
Would you mind adding `Deprecated: "...` properties to these ones that are being replaced / removed? As you probably already know, they will also need to be noted in the 5.0 Upgrade Guide: https://github.com/hashicorp/terraform-provider-azurerm/blob/main/website/docs/5.0-upgrade-guide.html.markdown